app/javascript/components/pages/Feed.js

```diff
@@ -2,46 +2,120 @@ import React from "react"
 import PropTypes from "prop-types"
 import { Nav, Button } from 'reactstrap'
 import { Link } from 'react-router-dom'
+import { Map as LeafletMap, TileLayer, Marker, Popup } from "react-leaflet"
+import { OpenStreetMapProvider } from 'leaflet-geosearch';
 
 
 class Feed extends React.Component {
-
+    constructor(props) {
+        super(props)
+        this.state = {
+          //hard-coded center of map
+          lat: 32.5732878512982,
+          lng: -117.107984521922,
+          zoom: 12,
+          //needs to be a number as an initial value
+          markerLat: 0,
+          markerLng: 0, 
+          markerPosition: [0,0],
+          markers: []
+        }   
+      }
+    //need this to load markers when switching between components
     componentDidMount() {
+        const { posts } = this.props
         this.props.changeSuccess(false)
+        const provider = new OpenStreetMapProvider();
+        posts.map((post)=>{
+            provider.search({ query: post.location })
+            .then((result) => { 
+                this.setState({markerLat:result[0].y, markerLng:result[0].x})
+                let title = post.title
+                let description = post.description
+                let marker = {
+                    location: [this.state.markerLat, this.state.markerLng], 
+                    title: title, 
+                    description: description
+                }
+                this.setState({markers: [...this.state.markers, marker ]})
```

Comment:
Why are we setting state on markerLat, and markerLng?  They are getting reset, and reset again, for every post.